cpp/src/arrow/CMakeLists.txt

```diff
@@ -126,6 +126,8 @@ set(ARROW_SRCS
     io/buffered.cc
     io/compressed.cc
     io/file.cc
+    io/hdfs.cc
```

Comment:
Should this be conditional on `ARROW_HDFS`?

Reply:
I don't know. @wesm What do you think?

Reply:
There are no additional build-time dependencies to compile this, so I think it's OK to have it here